### DISCUSSION TITLE
Residual of the weak form of the equation

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#23401] [Q&A General] [answered]
Hello everyone,
 
I am trying to add a source term to the navier-stokes equation. In fact, I am going to add the below term:

w*grad(c)

Here both w and c are scalar variables, so grad(c) and w*grad(c) are both vectors. How can I compute two residuals for this equation? Is there any similar MOOSE kernel that computes the residuals for similar terms?

Thanks,
Ali

**Comment by GiudGiud** [comment]
Hello

So this is a source term in the momentum equation?
Which Navier Stokes discretization are you using? (finite element, but which prefix? INS? INSAD? INSFE?)

Guillaume

**Reply by alimostafavi24** [reply]
w and c are both coupled variables from CH equation.

Yes, this is a source term for the momentum equation and I am using INS in 2D.

Regards,
Ali

**Reply by GiudGiud** [reply, marked as answer]
Ok great. You should be able to modify the coupledForce kernel to get what you want. 

You ll need to add a kernel for each component of the momentum equations, so two kernels
